internal/v2/controller/http/discovery.go

```diff
@@ -0,0 +1,41 @@
+package http
+
+import (
+	"net/http"
+
+	sdkCommon "github.com/edgexfoundry/device-sdk-go/internal/common"
+	"github.com/edgexfoundry/device-sdk-go/internal/container"
+	"github.com/edgexfoundry/device-sdk-go/internal/handler"
+	edgexErr "github.com/edgexfoundry/go-mod-core-contracts/errors"
+	contract "github.com/edgexfoundry/go-mod-core-contracts/models"
+)
+
+func (c *V2HttpController) Discovery(writer http.ResponseWriter, request *http.Request) {
+	ds := container.DeviceServiceFrom(c.dic.Get)
+	correlationID := request.Header.Get(sdkCommon.CorrelationHeader)
+
+	err := checkServiceLocked(request, ds.AdminState)
+	if err != nil {
+		c.sendError(writer, request, edgexErr.KindServiceLocked, "Service Locked", err, sdkCommon.APIV2DiscoveryRoute, correlationID)
+		return
+	}
+
+	if ds.OperatingState == contract.Disabled {
```

Comment:
In V2 API, we decided to remove OperatingState from Device Service, so please remove this section